runtime/tools/python/ttrt/runtime/module.cpp

```diff
@@ -136,4 +136,16 @@ PYBIND11_MODULE(_C, m) {
         os << env;
         return os.str();
       });
+
+  /**
+   * Cleanup code to force a well ordered destruction w.r.t. the GIL
+   *
+   * TODO: `clangd` reports that `capsule` is deprecated, figure out how to do
+   * this canonically
+   */
+  static int unused; // the capsule needs something to reference
+  py::capsule cleanup(&unused, [](PyObject *) {
+    ::tt::runtime::debug::Hooks::get().unregisterHooks();
+  });
+  m.add_object("_cleanup", cleanup);
```

Comment:
Re deprecated, it looks like only this variation of the capsule constructor is deprecated.  It looks like we can avoid the deprecated constructor and more succinctly written ([from their official docs](https://pybind11.readthedocs.io/en/stable/advanced/misc.html#module-destructors)):
```c++
auto cleanup_callback = []() {
    ::tt::runtime::debug::Hooks::get().unregisterHooks();
};
m.add_object("_cleanup", py::capsule(cleanup_callback));
```